packages/react/src/components/Icon/index.tsx

```diff
@@ -8,24 +8,45 @@ export interface IconProps extends React.HTMLAttributes<HTMLDivElement> {
   type: string;
 }
 
-const Icon = ({ label, type, className, ...other }: IconProps) => {
+function Icon({ label, className, type, ...other }: IconProps) {
+  const [, name, direction] = type.match(/(.*)-(right|left|up|down)$/) || [
+    '',
+    type
+  ];
+  const [IconSVG, setIcon] = useState<React.ComponentType<any> | null>(null);
+
+  useEffect(() => {
+    import(`./icons/${name}.svg`)
```

Comment:
What's wrong with `import`ing all of the icons and tree-shaking what we don't use out?

Reply:
Dynamically importing means adding new icons is mostly drag and drop without having to modify the main `<Icon>` component. What's your concern here?

Reply:
Manually importing new icons and wrapping them with `<Icon>` just to import seems like unnecessary overhead when we can let rollup do the heavy lifting.

Reply:
I understand that manually `import`ing all of them will be a pain, but I'm wondering how frequently we'll add/remove icon SVGs.

My main concern is dependent packages - are the consumers of `cauldron` setup to handle these `import`s?

Reply:
Yup! Rollup and dynamic-import-variables handles this for us, it outputs a map like this inside of `<Icon>`:

```js
function __variableDynamicImportRuntime0__(path) {
   switch (path) {
     case './icons/arrow.svg': return Promise.resolve().then(function () { return { 'default': require('./icons/arrow.svg.js') }; });
     case './icons/bolt.svg': return Promise.resolve().then(function () { return { 'default': require('./icons/bolt.svg.js') }; });
     case './icons/checkbox-checked.svg': return Promise.resolve().then(function () { return { 'default': require('./icons/checkbox-checked.svg.js') }; });
     /* ... and the rest of the icons */
     default: return Promise.reject(new Error("Unknown variable dynamic import: " + path));
   }
}

function Icon(_a) {
    React.useEffect(function () {
        __variableDynamicImportRuntime0__("./icons/" + name + ".svg")
            .then(function (icon) {
            setIcon(function () { return icon["default"]; });
        })["catch"](function (ex) {
            console.error("Could not find icon type \"" + type + "\".");
            setIcon(null);
        });
    }, [type])
  ...
}
```

As far as dependent packages are concerned, they don't even know there's a dynamic import. Assuming a bundler is used downstream it becomes an inline require.

Reply:
Awesome!!